runtime/common/src/migrations.rs

```diff
@@ -58,6 +62,73 @@ where
 		pallet_referenda::migration::v1::MigrateV0ToV1::<T>::post_upgrade(state)
 	}
 }
+pub struct UpdateFirstRoundRelayBlockNumber<T>(pub PhantomData<T>);
+impl<T> Migration for UpdateFirstRoundRelayBlockNumber<T>
+where
+	T: pallet_parachain_staking::Config,
+	T: cumulus_pallet_parachain_system::Config,
+	T: frame_system::Config,
+	u32: From<<<<T as frame_system::Config>::Block as Block>::Header as Header>::Number>,
+{
+	fn friendly_name(&self) -> &str {
+		"MM_UpdateFirstRoundRelayBlockNumber"
+	}
+
+	fn migrate(&self, _available_weight: Weight) -> Weight {
+		// First, fetch the last parachain block
+		let para_block: u32 = frame_system::Pallet::<T>::block_number().into();
+
+		// Fetch Round storage before the migration
+		let round_info = pallet_parachain_staking::Pallet::<T>::round();
+
+		// Calculate how many blocks have passed so far in the current round
+		let para_block_diff = para_block.saturating_sub(round_info.first);
+
+		// Calculate the percentage of the round so far (before the migration)
+		let percentage = (para_block_diff)
+			.saturating_mul(100)
+			.saturating_div(round_info.length);
+
+		// Calculate how many blocks should we substract from LastRelayChainBlockNumber
+		// given the new duration (round_info.length * 2) to have a first relay block of the
+		// round that corresponds with the percentage calculated in the step above.
+		let _new_block_diff = percentage
+			.saturating_mul(round_info.length * 2)
+			.saturating_div(100);
+
+		// Perform substraction
+		// TODO: fix with cherry-pick
```

Comment:
Reminder for this

Reply:
Now that we are not using the `LastRelayChainBlockNumber` storage, I think the current maths for the migration are no longer necessary. We should compute something for the round.first to be close to the current relay slot number. To give an example, the current babe slot in Polkadot is `283,961,413`.